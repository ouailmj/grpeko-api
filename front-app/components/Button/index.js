import React from 'react';
import PropTypes from 'prop-types';

const modalRoot = document.getElementById('modal-root');

class Button extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            labelText: 'Button'
        };
    }
    render() {
        return (
            <button  onClick={this.props.clickAction} className={this.props.classes} > {this.props.labelText} </button>
        );
      }
}

Button.propTypes = {
    labelText: PropTypes.string,
    clickAction: PropTypes.func,
    classes: PropTypes.string,
};

export default Button;
