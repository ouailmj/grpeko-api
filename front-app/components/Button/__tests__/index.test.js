import React from 'react';
import Button from '../index';
import renderer from 'react-test-renderer';

test('Test button', () => {
    const component = renderer.create(
        <Button labelText='Just Button' classes='btn-primary btn-test' clickAction={(e)=> {}}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
