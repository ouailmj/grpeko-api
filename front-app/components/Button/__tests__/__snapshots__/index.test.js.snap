// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button 1`] = `
<button
  className="btn-primary btn-test"
  onClick={[Function]}
>
   
  Just Button
   
</button>
`;
